wallet/api/services_manager.go

```diff
@@ -49,7 +51,13 @@ func (ns *ServicesManager) RegisterService(network, url string, sessions *sessio
 		if err != nil {
 			return fmt.Errorf("could not retrieve the wallet %q associated to the token %q: %w", tokenInfo.Wallet.Name, token.Token, err)
 		}
-		if err := sessions.ConnectWalletForLongLivingConnection(token.Token, w); err != nil {
+
+		var expiry *time.Time
+		if tokenInfo.Expiry != nil {
+			expiry = ptr.From(time.Unix(*tokenInfo.Expiry, 0))
+		}
```

Comment:
This check should be delegated to the `sessions.ConnectWalletForLongLivingConnection()` as it's its responsability to check valid input, _ultimately_.

If we are to call `sessions.ConnectWalletForLongLivingConnection()` somewhere else we don't want to copy the expiry verification.